cpp/comms/std/src/ucp_helper.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ucp/api/ucp.h>
+#include <ucp/api/ucp_def.h>
+
+#include <utils.h>
+static const ucp_tag_t default_tag_mask = -1;
+
+/**
+ * @brief Continues progress on worker until a message request has completed
+ */
+static void wait(ucp_worker_h ucp_worker, struct ucx_context *context) {
+  while (context->completed == 0) {
+    ucp_worker_progress(ucp_worker);
+  }
+}
+
+/**
+ * @brief callback for flushing worker
+ */
+static void flush_callback(void *request, ucs_status_t status) {}
+
+/**
+ * @brief Flush the send buffer on the given endpoint for the given worker
+ */
+static ucs_status_t flush_ep(ucp_worker_h worker, ucp_ep_h ep) {
+  void *request;
+
+  request = ucp_ep_flush_nb(ep, 0, flush_callback);
+  if (request == NULL) {
+    return UCS_OK;
+  } else if (UCS_PTR_IS_ERR(request)) {
+    return UCS_PTR_STATUS(request);
+  } else {
+    ucs_status_t status;
+    do {
+      ucp_worker_progress(worker);
+      status = ucp_request_check_status(request);
+    } while (status == UCS_INPROGRESS);
+    ucp_request_release(request);
```

Comment:
Is `ucp_request_release` a call that you've implemented? UCP API only has [`ucp_request_free`](https://github.com/openucx/ucx/blob/master/src/ucp/api/ucp.h#L2898)

Reply:
Also worth noting that context->completed is not set to 0 by free. This happens only at request init. More [here](https://github.com/openucx/ucx/blob/master/src/ucp/api/ucp.h#L665)

Reply:
Actually, it looks like this `flush_ep` function (which I took from the original `hello_world.c` example, might be deprecated now. Here's the [reference](http://openucx.github.io/ucx/api/v1.5/html/group___u_c_p___w_o_r_k_e_r.html#gac461d1034a3313612243a5829b7e3847)